Source/Core/Check.js

```diff
@@ -0,0 +1,144 @@
+/*global define*/
+define([
+    './defaultValue',
+    './defined',
+    './DeveloperError',
+    './isArray'
+    ], function(
+        defaultValue,
+        defined,
+        DeveloperError,
+        isArray) {
+    'use strict';
+
+    /**
+     * Contains functions for checking that supplied arguments are of a specified type
+     * or meet specified conditions
+     * @private
+     *
+     * @exports Check
+     */
+    var Check = {};
+
+    Check.typeOf = {};
+    Check.numeric = {};
+
+    var errors = {
+        defined: getUndefinedErrorMessage,
+        failedType: getFailedTypeErrorMessage
+    };
+
+    function getUndefinedErrorMessage(name) {
+        return defaultValue(name, 'a value') + ' is required but is undefined.';
+    }
+    function getFailedTypeErrorMessage(actual, expected) {
+        return 'Expected ' + expected + ', got ' + actual;
+    }
+
+    /**
+     * @param {} test The value that is to be checked
```

Comment:
Can you look at other doc to see if `{}` is correct here?  I think it should be `{Any}` or `{*}` but this might have changed at some point.

Also, we usually don't document functions for `@private` classes, but I _do_ think it is useful to document these since I don't think we will memorize them all and also new contributors will be working with them.